### ISSUE TITLE
<iframe> with use_native: true does not lazy load

### ISSUE THREAD

**Issue by tyteen4a03** [verlok/lazyload#442]
**Describe the bug**
If `use_native` is on, with a browser that supports lazy loading out of the box, no `<iframe>`s are lazy-loaded. 

**To Reproduce**
1. Create a web page with a lot of `<iframe>`s (Facebook embedded posts is a good candidate). I used `<iframe data-src="...">` but adding `loading="lazy"` does not change the behaviour.
2. Use LazyLoad with selector `<iframe>` and `use_native: true`.
3. Load the web page and watch the Network tab. It should indicate that the browser is loading all tabs at once.
4. Turn off `use_native` and reload the web page; the network usage should be back to normal.

**Expected behavior**
`use_native: true` should work on `<iframe>`s.

**LazyLoad version**
Please report which version of LazyLoad you're using.
- Version 15.1

**Desktop (please complete the following information):**
 - OS: macOS 10.15.4
 - Browser: Firefox 75, Chrome 83

**Comment by verlok** [comment]
Hey @tyteen4a03 thank you for reporting. I'm investigating.

**Comment by verlok** [comment]
I've verified that this happens just like you reported. I'm working on it.

**Comment by verlok** [comment]
Here's an update. 

I've tested [the native lazyloading demo](https://www.andreaverlicchi.eu/lazyload/demos/native_lazyload.html) which uses NO JS, and everything works as expected: the iframes are lazily loaded as you scroll down, even if they are loaded way before you scroll down to them (the browser must start loading them sooner because they usually involve many other network requests).

I've tested [the hybrid lazyloading demo](https://www.andreaverlicchi.eu/lazyload/demos/native_lazyload_conditional.html) which uses this LazyLoad library and `use_native` and I've found out that in that case the `iframe`s are loaded as soon as the page is loaded.

I've checked LazyLoad's code and it goes "if native lazy loading is supported in the browser, set the `loading=lazy` attribute in the matched elements, then set the `src` element". So everything looked correct.

So I started thinking that the immediate loading of the `iframe`s must be triggered by the fact that I'm applying the `src` in the code, and the browser is ignoring the `loading=lazy` attribute when this is done directly. 

What I did is to create [this Codepen](https://codepen.io/verlok/pen/mderGoE?editors=1000) to test **native lazy loading** on `iframe`s and isolate the problem. At the bottom of it, there are 4 `iframe`s with the `loading="lazy"` attribute, like the following.

```html
<iframe loading="lazy" src="{{facebook post embed URL}}" width="552" height="513" style="border:none;overflow:hidden" scrolling="no" frameborder="0" allowTransparency="true" allow="encrypted-media"></iframe>
```

What I did notice is that the browser acts differently depending on where the page it's run:

- inside the codepen editor, the iframes are loaded lazily as you scroll down, as expected
- as a new page (I used codepen's debug mode, I guess you can download the pen and execute it in your browser) the iframes are loaded as soon as you land, as expected

So I'm guessing the decision whether or not to lazily load the `iframe`s is taken by the browser, maybe depending on the network traffic, e.g. on how many requests the page already made.

Now I'm going to fork it and see what happens when the `src` attributes are set via Javascript.

**Comment by verlok** [comment]
(by the way, I've also found this...)

![image](https://user-images.githubusercontent.com/1127721/79811822-58a86800-8377-11ea-99bb-d951762a0918.png)

**Comment by tyteen4a03** [comment]
Interesting find. I definitely don't want the browser deciding whether to lazy load as my webpage can have up to a 100 <iframe>s and that's about 150MB of traffic...

**Comment by verlok** [comment]
I've forked it, [in this Codepen](https://codepen.io/verlok/pen/WNQGapy?editors=1010) I'm setting the `src` in JS. In this case, the loading of the `iframe`s starts right away, no matter what. 

So this considered, I guess that using hybrid native lazy loading on `iframe`s is not a best practice, and it should be discouraged. I definitely need to write this in the documentation!

**Comment by verlok** [comment]
> I definitely don't want the browser deciding whether to lazy load as my webpage can have up to a 100 <iframe>s and that's about 150MB of traffic...

Then you shouldn't use the `use_native` lazy loading to load `iframe`s.

**Comment by verlok** [comment]
If you still want to use the `use_native` option for your images, you could do something like:

```js
// Supposing your lazy images have a "lazy" class applied
const lazyContent = new LazyLoad({
    elements_selector: "img.lazy", 
    use_native: true // <-- native for images
});

// Supposing your lazy iframes have the class "lazy" applied
const lazyBackground = new LazyLoad({
    elements_selector: "iframe.lazy"
    // DON'T PASS use_native: true HERE
});
```

What do you think?

**Comment by tyteen4a03** [comment]
Oh yes that's the solution I ended up using before reporting the bug.

**Comment by verlok** [comment]
I’m glad you solved. 
And thank you again for reporting, to investigate and discuss is always a learning occasion.

**Comment by verlok** [comment]
Closed this in [12c5005](https://github.com/verlok/lazyload/commit/12c5005142bd5fd7f3b6be8ace83a1289bae057e)
